bindings/pyroot/pythonizations/test/rdf_define_pyz.py

```diff
@@ -70,6 +70,43 @@ def norm(x_arr):
         arr = np.sqrt(arr*arr + arr*arr )
         flag = np.array_equal(rdf.AsNumpy()["mag"], arr)
         self.assertTrue(flag)
+
+    def test_cpp_functor(self):
+        """
+        Test that a C++ functor can be passed as a callable argument of a
+        Define operation.
+        """
+
+        ROOT.gInterpreter.Declare("""
+        struct MyFunctor
+        {
+            ULong64_t operator()(ULong64_t l) { return l*l; };
+        };
+        """)
+        f = ROOT.MyFunctor()
+
+        rdf = ROOT.RDataFrame(5)
+        rdf2 = rdf.Define("x", f, ["rdfentry_"])
+
+        for x,y in zip(rdf2.Take['ULong64_t']("rdfentry_"), rdf2.Take['ULong64_t']("x")):
```

Comment:
I think both of the `Take` calls are missing a `GetValue`. Also, by default they return a `std::vector<ULong64_t>` and I don't think `x*x` can be done with those.

Reply:
Probably we want to book both operations like `rdf2.Take['ULong64_t', 'RVec<ULong64_t>']("rdfentry_")` and then call `GetValue` in the zip

Reply:
Actually that might not be necessary because the fact of iterating over the RResultPtr of vector unwraps the individual integers, for example:
```python
import ROOT

ROOT.gInterpreter.Declare("""
std::function<ULong64_t(ULong64_t)> myfun = [](ULong64_t l) { return l*l; };
""")

rdf = ROOT.RDataFrame(5)
rdf2 = rdf.Define("x", ROOT.myfun, ["rdfentry_"])

for x,y in zip(rdf2.Take['ULong64_t']("rdfentry_"), rdf2.Take['ULong64_t']("x")):
    print(type(x), type(y))
```
This prints all integers.

Reply:
:exploding_head: this should be better documented. Also, does this evaluate both `Take` operations in one graph or are they run separately?

Reply:
Ok I tried apparently it runs only once, amazing!

Reply:
Yes, they run only once because they are both first booked (when constructing the arguments to zip) and then inside zip they are iterated on, which triggers the loop once.